app/controllers/candidate_interface/personal_details/languages_controller.rb

```diff
@@ -21,13 +21,21 @@ def create
 
       def edit
         @languages_form = LanguagesForm.build_from_application(current_application)
+
+        @return_to = if redirect_back_to_application_review_page?
+                       application_review_path_and_params
+                     else
+                       { path: candidate_interface_personal_details_complete_path, params: {} }
+                     end
       end
 
       def update
         @application_form = current_application
         @languages_form = LanguagesForm.new(languages_params)
 
         if @languages_form.save(current_application)
+          return redirect_to candidate_interface_application_review_path if redirect_back_to_application_review_page?
```

Comment:
For each of these cases, could the return_to path that we resolve in the `edit` action be hidden fields in the form? Could potentially do just the one call here then:

`redirect_to params.fetch(return_to_path)`

I might be missing something, but I think that means you then don't need to handle the `return-to` param in the edit template.

Reply:
That's an interesting one - I had a play around with this idea but I think it added more complexity. When a candidate visits the application review page and decides to update a field (i.e. by clicking 'Change') we need to past the 'return_to' as query params. By sticking with query params the whole way through it may make the code a little cleaner rather than having a mix of query params and hidden fields (?).